rest/route/github.go

```diff
@@ -370,26 +373,45 @@ func (gh *githubHookApi) handleGitTag(ctx context.Context, event *github.PushEve
 	}
 	catcher := grip.NewBasicCatcher()
 	for _, pRef := range projectRefs {
-		// if a version for this revision exists for this project, add tag
-		existingVersion, err := gh.sc.FindVersionByProjectAndRevision(pRef.Id, hash)
-		if err != nil {
-			catcher.Add(errors.Wrapf(err, "problem finding version for project '%s' with revision '%s' to add tag '%s'",
-				pRef.Id, hash, tag.Tag))
-			continue
+		var existingVersion *model.Version
+		noExistingVersionMessage := message.Fields{
+			"source":  "github hook",
+			"message": "",
+			"ref":     event.GetRef(),
+			"event":   gh.eventType,
+			"project": pRef.Id,
+			"branch":  pRef.Branch,
+			"owner":   pRef.Owner,
+			"repo":    pRef.Repo,
+			"hash":    hash,
+			"tag":     tag,
 		}
-		if existingVersion == nil {
-			grip.Debug(message.Fields{
-				"source":  "github hook",
-				"message": "no version to add tag to",
-				"ref":     event.GetRef(),
-				"event":   gh.eventType,
-				"project": pRef.Id,
-				"branch":  pRef.Branch,
-				"owner":   pRef.Owner,
-				"repo":    pRef.Repo,
-				"hash":    hash,
-				"tag":     tag,
-			})
+		err = util.Retry(
+			ctx,
+			func() (bool, error) {
+				// If a version for this revision exists for this project, add tag
+				// Retry in case a commit and a tag are pushed at around the same time, and the version isn't ready yet
+				existingVersion, err = gh.sc.FindVersionByProjectAndRevision(pRef.Id, hash)
+				if err != nil {
+					catcher.Add(errors.Wrapf(err, "problem finding version for project '%s' with revision '%s' to add tag '%s'",
```

Comment:
Thinking about it now, it seems odd to add errors to a catcher here, but not to consider the error returned from util.Retry at all. Also, an error here would be very unusual, i.e. something weird happened with the DB, and there's no point in us adding that error to the catcher 5 separate times. 

I would suggest just returning the errors, and then after util.Retry catching the error, and then both the err and the existingVersion case will actually just get hit by the "errors updating/creating versions for git tag" logging at the end.